### PR TITLE
[Design2-185-Checkbox] Checkbox Component - DSCO - Tests - Revisit all the tests, rewrite jsx tests to tsx, fix linter ignoring comments (get rid of data-testIds)

### PR DESCRIPTION
**Pull request by levadadenys**
### [Design2-185-Checkbox] Checkbox Component - DSCO - Tests - Revisit all the tests, rewrite jsx tests to tsx, fix linter ignoring comments (get rid of data-testIds)

- rewritten `CheckboxTest` in typescript
- refactored/optimized `CheckboxTest.tsx`
- added more tests for `Checkbox` component

This PR only updates existing tests and add couple new ones. No visual changes are intended by this PR.

## Links
- jira ticket: part of [Design2-185](https://codedotorg.atlassian.net/browse/DESIGN2-185)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/componentLibrary/CheckboxTest.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    expect(spyOnChange).toHaveBeenCalledTimes(1);
+    expect(spyOnChange).toHaveBeenCalledWith(true);
+
+    await user.click(checkbox);
+    expect(checkbox.checked).toBe(false);
+    expect(spyOnChange).toHaveBeenCalledTimes(2);
+    expect(spyOnChange).toHaveBeenCalledWith(false);
+  });
+
+  it('handles indeterminate state correctly', async () => {
```
Comment (bencodeorg) [inline apps/test/unit/componentLibrary/CheckboxTest.tsx:83]: Nit: "handles initial indeterminate state correctly"? I was just doing some Googling, for my learning, what is the use case for the "indeterminate" option?

**Review by bencodeorg — APPROVED** [review]
Git didn't get a clean diff between the jsx and tsx versions, I threw them in a diffchecker. LGTM, nice refactor!

**Review by sanchitmalhotra126 — APPROVED** [review]

File: apps/test/unit/componentLibrary/CheckboxTest.tsx (hunk 1 above)
Comment (levadadenys) [reply apps/test/unit/componentLibrary/CheckboxTest.tsx:83]: @bencodeorg sorry for long response

> "handles initial indeterminate state correctly"

Not sure why add initial here. Can you explain please?

> what is the use case for the "indeterminate" option?

It's something like 'in bettween'. A good example here is a tree of checkboxes:

```
[-] Folder 1
    [v] opt a
    [ ] opt b
[ ] Folder 2
    [ ] opt a
```

[v] - selected checkbox, [-] indeterminate checkbox

Here's more about it on [MDN DOCS](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes)


> There are not many use cases for this property. The most common is when a checkbox is available that "owns" a number of sub-options (which are also checkboxes). If all of the sub-options are checked, the owning checkbox is also checked, and if they're all unchecked, the owning checkbox is unchecked. If any one or more of the sub-options have a different state than the others, the owning checkbox is in the indeterminate state.

Comment (bencodeorg) [reply apps/test/unit/componentLibrary/CheckboxTest.tsx:83]: Got it, thanks! Re: test name, never mind, the name is making sense to me at second glance.

**Pull request merged**